[clydelhui] iP

clydelhui opened the pull request:
# Duke
> "How much wood would a woodchuck chuck if a woodchuck could chuck wood? As much as the number of tasks that Duke can store for you." - Clyde Lhui

Duke stores everything for you! It's
- Fantastic
- Awesome
- Written by a cool guy 😸 
- ~~Great~~ THE ABSOLUTE GREATEST

To experience the greatest of Duke,
1. Download it from [here](https://github.com/clydelhui/ip/releases/tag/v0.1)
2. Click on it
3. Experience the wonders of Duke!

Features:
[x] Task management
[ ] Search (coming soon)

If you are a Java programmer, you can take a look at the source code too!. Here is the `main` method in Duke!

```java
public static void main(String[] args) {


        Scanner usrInput = new Scanner(System.in);
        Ui ui = new Ui(usrInput);

        ui.welcome();

        Storage storage = new Storage("data.txt", ui);

        TaskList taskList = new TaskList(storage, ui);

        Parser parser = new Parser();

        String input = "";

        while (true) {
            input = ui.getInput();
            try {
                Command command = parser.parse(input);
                command.execute(taskList, ui, storage);
            } catch (DukeException e) {
                ui.errorDisplay(e);
            }
        }

    }
```

Commit:
Add Gradle support

clydelhui committed:
Modified Duke.java

clydelhui committed:
Modified Duke.java

clydelhui committed:
Added Task.java and Modified Duke.java

clydelhui committed:
Added new task types, modified Duke.java

clydelhui committed:
Added Text UI Testing

clydelhui committed:
Mofiy runtest.bat to compile packages, created exceptions to handle invalid inputs

clydelhui committed:
Modify Duke.java to include delete functionality

clydelhui committed:
Add saving functionality, add CommandHandler and TaskList classes

clydelhui committed:
Change type of Deadline to LocalDate

clydelhui committed:
Merge branch 'branch-Level-7'

clydelhui committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

0x787af25e reviewed (commented):
Your code is of good quality (though I see some parts that do not conform to the coding standard).

zichen-3974 commented on src/main/java/CommandHandler.java:
```diff
@@ -0,0 +1,71 @@
+        this.taskList = taskList;
+    }
+
+    public void execute(String command) throws IllegalCommandException,
```
Perhaps you can consider having Javadocs for this method

zichen-3974 commented on src/main/java/CommandHandler.java:
```diff
@@ -0,0 +1,71 @@
+    public void execute(String command) throws IllegalCommandException,
+            IllegalInputException, NumberFormatException {
+        String[] parsedCommand = command.split(" ", 2);
+        switch (parsedCommand[0]) {
```
I like your implementation! Perhaps you can consider further abstraction, e.g having an abstract Command class that different types of commands can extend from :)

zichen-3974 reviewed (commented):
Nice job! Your code is clean and easy to follow overall, and your variables are named meaningfully. You may want to consider adding some Javadocs in the future :) 👍

YZmunchmunch commented on src/main/java/items/Deadline.java:
```diff
@@ -1,13 +1,15 @@
     }
 
-    public Deadline(String description, boolean done, String endDate) {
+    public Deadline(String description, boolean done, LocalDate endDate) {
```
perhaps the parameter "done" could be changed to "isDone" instead!

JThh commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+    }
+
+    public void delete(int index){
+        Task deleted = this.taskList.remove(index - 1);
```
Can here be some input validity check to ensure index is never less than 1?

JThh commented on src/main/java/dukeexceptions/IllegalInputException.java:
```diff
@@ -0,0 +1,12 @@
+package dukeexceptions;
```
Please conform to the camelNaming rule here to name the package.

JThh reviewed (commented):
The codes are easy to read and well-functional.

clydelhui committed:
Add Ui, Storage, Parser, and Command classes.
Add additional exceptions with better handling.

clydelhui committed:
Merge remote-tracking branch 'origin/add-gradle-support'

clydelhui committed:
Add checkstyle to gradle, fix style issues, add JUnit tests for AddCommand and Parser classes.

clydelhui committed:
Add new StorageException, JavaDocs

clydelhui committed:
Add new VoidCommand type "find"

clydelhui committed:
Modify files to fit coding standard

clydelhui committed:
Merge branch 'branch-A-JavaDoc'

clydelhui committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/commands/VoidCommand.java

clydelhui committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/elems/TaskList.java

clydelhui committed:
add UI with JavaFX

clydelhui committed:
Merge branch 'branch-Level-10'

clydelhui committed:
add assertions

When TaskList methods are called, the ArrayList of Tasks should never be
null.

Adding assertions allows us to ensure that this invariant is always
true.

Let's add assertions to the TaskList methods to ensure that all TaskList
Objects which are targets for method invocation have non-null ArrayLists
at the time of method calling.

clydelhui committed:
Improve code quality

execute method in the AddCommmand class is long and can be hard to debug
 in the future. If more commands are added, the code will get even
longer.

Let's abstract out the steps taken in each case into a separate method.

Future additions of AddCommands can adopt the same style of execution.

clydelhui committed:
Merge pull request #2 from clydelhui/branch-A-Assertions

add assertions

clydelhui committed:
Merge branch 'master' of https://github.com/clydelhui/ip

* 'master' of https://github.com/clydelhui/ip:
  add assertions

clydelhui committed:
Merge pull request #3 from clydelhui/branch-CodeQuality

Improve code quality

clydelhui committed:
add sort functionality, empty list exception

The user may wish to sort the task list by description to make it easy to
read.

Let's add a sorting feature to enable the user to do this.

Some commands such as list and sort will access the list elements. If
the list is empty, they are unable to work as intended.

Let's add an exception to show the case when an empty list is being
accessed by commands that work with the whole list.

clydelhui committed:
Merge branch 'branch-BCD-Extension'

clydelhui committed:
update README.md, make minor bug fixes

clydelhui committed:
Modify README.md
